packages/app/src/systems/Chains/eth/containers/EthAssetsDialog.tsx

```diff
@@ -1,52 +1,193 @@
+  const [editable, setEditable] = useState(false);
+
+  const form = useSetAddressForm();
+
+  const assetQuery = useWatch({ name: 'address', control: form.control });
+
+  const {
+    assets,
+    handlers,
+    showCustomTokenButton,
+    showUseTokenButton,
+    assetInfo,
+    isLoading,
+    doesAssetExist,
+  } = useManageEthAssets({ assetQuery });
+
+  const onSubmitToken = () => {
+    handlers.addAsset({
+      asset: {
+        address: assetInfo?.address,
+        decimals: assetInfo?.decimals,
+        symbol: assetInfo?.symbol,
+      },
+    });
+    form.resetField('address');
+  };
+
+  const onSubmitCustomToken = (data: SetAddressFormValues) => {
+    store.openAddAssetsDialog({
+      assetAddress: data.address,
+    });
+  };
 
   return (
     <>
-      <Dialog.Heading css={styles.dialogHeading}>
-        <Box.Flex justify="space-between" css={styles.dialogHeadingContainer}>
+      <Dialog.Close />
+      <Dialog.Heading>
+        <Box.Flex gap="$4" justify="start">
+          {editable && (
+            <IconButton
+              aria-label="Set editable to false"
+              variant="link"
+              icon="ArrowLeft"
+              onPress={() => setEditable(false)}
+            />
+          )}
           <Text color="intentsBase12" fontSize="sm">
-            Select token
+            {!editable ? 'Select token' : 'Manage token list'}
           </Text>
-          <Dialog.Close />
         </Box.Flex>
       </Dialog.Heading>
-      <Dialog.Description>
-        <CardList isClickable>
-          {assets.map((asset, i) => (
-            <CardList.Item
-              key={asset.address + asset.symbol + String(i)}
-              onClick={() =>
-                bridgeHandlers.changeAssetAddress({
-                  assetAddress: asset.address,
-                })
-              }
-            >
-              {asset.symbol}
-            </CardList.Item>
-          ))}
+      <Dialog.Description css={styles.dialogDescription}>
+        <Box.Flex align="center" css={styles.controllerWrapper}>
+          <Controller
+            name="address"
+            control={form.control}
+            render={(props) => {
+              return (
+                <Form.Control css={{ width: '$full' }}>
+                  <Input size="md" css={styles.headerInput}>
+                    <Input.Field
+                      {...props.field}
+                      placeholder="Search or paste custom address"
+                    />
+                    {isLoading && (
+                      <Input.ElementRight>
+                        <Spinner />
+                      </Input.ElementRight>
+                    )}
+                  </Input>
+                  {!doesAssetExist &&
+                    !(showCustomTokenButton || showUseTokenButton) &&
+                    !isLoading && (
+                      <Form.HelperText>{`No asset found for your search "${assetQuery}"`}</Form.HelperText>
```

Comment:
AssetList should be hidden when this is shown.

if there's no asset for the search, no asset should be show in the list